internal/api/v1beta1/permission_check.go

```diff
@@ -22,6 +22,76 @@ func (h Handler) CheckResourcePermission(ctx context.Context, req *shieldv1beta1
 	//	logger.Error(formattedErr.Error())
 	//	return nil, status.Errorf(codes.NotFound, formattedErr.Error())
 	//}
+	//  To have backward compatibility
+	if req.ObjectId != "" && len(req.ResourcePermissions) == 0 {
+		return h.checkSingleResourcePermission(ctx, req)
+	}
+
+	if len(req.ResourcePermissions) > h.maxBulkCheckLimit || len(req.ResourcePermissions) == 0 {
+		formattedErr := fmt.Errorf("%s: %s", ErrRequestBodyValidation, "resource_permissions")
+		logger.Error(formattedErr.Error())
+		return nil, formattedErr
+	}
+
+	var results []*shieldv1beta1.CheckResourcePermissionResponse_ResourcePermissionResponse
+	bulkCheckCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	resultCh := make(chan *shieldv1beta1.CheckResourcePermissionResponse_ResourcePermissionResponse)
```

Comment:
why was `shieldv1beta1.CheckResourcePermissionResponse_ResourcePermissionResponse` used to make chan?

Reply:
Result chan needs to have the context of the resource in addition to the actual response(allowed or not). Hence utilized `shieldv1beta1.CheckResourcePermissionResponse_ResourcePermissionResponse` to collect that information and appended it.

Reply:
can we make a struct and then transform it into `shieldv1beta1.CheckResourcePermissionResponse_ResourcePermissionResponse` ?

Reply:
Modified @krtkvrm